core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -711,6 +716,57 @@ public void setWorkingDirectory(Path path) {
     }
   }
 
+  /**
+   * Check if the input URI is Alluxio on Zookeeper URI.
+   * If it is, set Zookeeper configuration in Hadoop configuration.
+   *
+   * @param uri a uri that may contain Zookeeper information
+   * @param conf hadoop configuration to set
+   */
+  private static void checkAndSetZookeeperConfiguration(URI uri,
```

Comment:
This only supports zk-in-URI for hdfs API - to support it for native API as well (e.g. shell), we need to put the parsing in `AlluxioURI` instead.